src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Views/Resource.sql

```diff
@@ -0,0 +1,182 @@
+          ,TransactionId
+      )
+    SELECT ResourceTypeId
+          ,ResourceSurrogateId
+          ,ResourceId
+          ,Version
+          ,IsDeleted
+          ,RequestMethod
+          ,RawResource
+          ,IsRawResourceMetaSet
+          ,SearchParamHash
+          ,TransactionId
+      FROM Inserted
+      WHERE IsHistory = 0
+
+  INSERT INTO dbo.ResourceHistory
+      (
+           ResourceTypeId
+          ,ResourceSurrogateId
+          ,ResourceId
+          ,Version
+          ,IsDeleted
+          ,RequestMethod
+          ,RawResource
+          ,IsRawResourceMetaSet
+          ,SearchParamHash
+          ,TransactionId
+          ,HistoryTransactionId
+      )
+    SELECT ResourceTypeId
+          ,ResourceSurrogateId
+          ,ResourceId
+          ,Version
+          ,IsDeleted
+          ,RequestMethod
+          ,RawResource
+          ,IsRawResourceMetaSet
+          ,SearchParamHash
+          ,TransactionId
+          ,HistoryTransactionId
+      FROM Inserted
+      WHERE IsHistory = 1
+END
+GO
+CREATE TRIGGER dbo.ResourceUpd ON dbo.Resource INSTEAD OF UPDATE
+AS
+BEGIN
+  IF UPDATE(SearchParamHash) AND NOT UPDATE(IsHistory)
+  BEGIN
+    UPDATE B
+      SET SearchParamHash = A.SearchParamHash -- this is the only update we support
+      FROM Inserted A
+           JOIN dbo.ResourceCurrent B ON B.ResourceTypeId = A.ResourceTypeId AND B.ResourceSurrogateId = A.ResourceSurrogateId
+      WHERE A.IsHistory = 0
+    
+    RETURN
+  END
+
+  IF NOT UPDATE(IsHistory)
+    RAISERROR('Generic updates are not supported via Resource view',18,127)
+
+  DELETE FROM A
+    FROM dbo.ResourceCurrent A
+    WHERE EXISTS (SELECT * FROM Inserted B WHERE B.ResourceTypeId = A.ResourceTypeId AND B.ResourceSurrogateId = A.ResourceSurrogateId AND B.IsHistory = 1)
+
+  INSERT INTO dbo.ResourceHistory
+      (
+           ResourceTypeId
+          ,ResourceSurrogateId
+          ,ResourceId
+          ,Version
+          ,IsDeleted
+          ,RequestMethod
+          ,RawResource
+          ,IsRawResourceMetaSet
+          ,SearchParamHash
+          ,TransactionId
+          ,HistoryTransactionId
+      )
+    SELECT ResourceTypeId
+          ,ResourceSurrogateId
+          ,ResourceId
+          ,Version
+          ,IsDeleted
+          ,RequestMethod
+          ,RawResource
+          ,IsRawResourceMetaSet
+          ,SearchParamHash
+          ,TransactionId
+          ,HistoryTransactionId
+      FROM Inserted
+      WHERE IsHistory = 1
+END
+GO
+CREATE TRIGGER dbo.ResourceDel ON dbo.Resource INSTEAD OF DELETE
+AS
+BEGIN
+  DELETE FROM A
+    FROM dbo.ResourceCurrent A
+    WHERE EXISTS (SELECT * FROM Deleted B WHERE B.ResourceTypeId = A.ResourceTypeId AND B.ResourceSurrogateId = A.ResourceSurrogateId AND B.IsHistory = 0)
+
+  DELETE FROM A
+    FROM dbo.ResourceHistory A
+    WHERE EXISTS (SELECT * FROM Deleted B WHERE B.ResourceTypeId = A.ResourceTypeId AND B.ResourceSurrogateId = A.ResourceSurrogateId AND B.IsHistory = 1)
```

Comment:
How are we handling a delete request to only delete historical versions? This looks like it would always delete the resource from both the current and historical table.

Reply:
No. We check IsHistory flag on the deleted virtual table, so we delete from appropriate physical table only.